### PR TITLE
DT-1887

### PR DESCRIPTION
**Pull request by siren**

### CODE REVIEW

**Comment by siren** [comment]
do not  merge to master before feature is complete.

there will be pr's that build on top of this

File: sass/base/_helper-classes.scss [hunk 1]
```diff
@@ -1,4 +1,4 @@
-/*
+.call/*
```
Comment (hannesj) [inline sass/base/_helper-classes.scss]: ?

File: app/util/legUtils.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+// converts the given parameter into a string in format HHmm
```
Comment (hannesj) [inline app/util/legUtils.js]: This description doesn't seem to match content

File: app/util/legUtils.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+ */
+export function isCallAgencyPickupType(leg) {
+  return filterLegStops(leg, stoptime => (stoptime.pickupType === 'CALL_AGENCY')).length > 0;
+}
+
+export function isCallAgencyDeparture(departure) {
+  return departure.pickupType === 'CALL_AGENCY';
+}
+
+export default null;
```
Comment (hannesj) [inline app/util/legUtils.js]: this is not needed

File: app/component/map/ItineraryLine.js [hunk 4]
```diff
@@ -90,53 +93,98 @@ class ItineraryLine extends React.Component {
...
+                position={{
+                  lat: middle.lat,
+                  lon: middle.lon,
+                }}
+                className="call"
+                icon="icon-icon_call"
+              />);
+
+            objs.push(
+              <StopMarker
```
Comment (hannesj) [inline app/component/map/ItineraryLine.js]: Aren't the stop markers added somewhere else

File: app/component/RouteNumberContainer.js [hunk 5]
```diff
@@ -3,7 +3,7 @@ import get from 'lodash/get';
 
 import RouteNumber from './RouteNumber';
 
-const getText = (route, config) => {
+const getText = (route, props, config) => {
```
Comment (hannesj) [inline app/component/RouteNumberContainer.js]: props is not used

File: app/component/ItineraryTab.js [hunk 6]
```diff
@@ -157,13 +157,23 @@ export default Relay.createContainer(ItineraryTab, {
...
+            }
           }
           trip {
             gtfsId
             tripHeadsign
             pattern {
               code
             }
+            stoptimes {
+              pickupType
```
Comment (hannesj) [inline app/component/ItineraryTab.js]: Should we add this directly to the leg information for the departure and destination stops?

File: app/component/CallAgencyLeg.js [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+              </span>
+            </div>
+          </div>
+        </div></div>);
+  }
+
+  render() {
+    return (
+      <div>
+        {[].concat([this.renderMain()])}
```
Comment (hannesj) [inline app/component/CallAgencyLeg.js]: Is there some reason for this?

File: app/component/CallAgencyLeg.js [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+        }
+      </span>];
+
+    const firstLegClassName = this.props.index === 0 ? ' start' : '';
+    const modeClassName = 'call';
+
+    return (
+      <div>
+        <div
+          key={this.props.index}
```
Comment (hannesj) [inline app/component/CallAgencyLeg.js]: key should be on the outermost element

File: app/component/CallAgencyLeg.js [hunk 9]
```diff
@@ -0,0 +1,158 @@
...
+              <Icon img="icon-icon_search-plus" className="itinerary-search-icon" />
+            </div>
+            <div className="itinerary-transit-leg-route call">
+              <span className="warning-message">
+                <FormattedMessage
+                  id="warning-call-agency"
+                  values={{
+                    routeName: <span className="route-name">{this.props.leg.route.longName}</span>,
+                  }}
+                  defaultMessage={`TODO Liikennöidään kutsujoukkoliikenteenä: {routeName},
```
Comment (hannesj) [inline app/component/CallAgencyLeg.js]: Change to english translation

File: app/component/CallAgencyLeg.js [hunk 10]
```diff
@@ -0,0 +1,227 @@
...
+import RouteNumber from './RouteNumber';
+import Icon from './Icon';
+import StopCode from './StopCode';
+import LegAgencyInfo from './LegAgencyInfo';
+
+class CallAgencyLeg extends React.Component {
+
+  stopCode = stopCode => stopCode && <StopCode code={stopCode} />;
+
+  renderMain = () => {
```
Comment (hannesj) [inline app/component/CallAgencyLeg.js]: This is not used anywhere

File: app/component/CallAgencyLeg.js [hunk 11]
```diff
@@ -0,0 +1,227 @@
...
+        {moment(this.props.leg.startTime).subtract(this.props.leg.departureDelay, 's')
+          .format('HH:mm')
+        }
+      </span>];
+
+    const firstLegClassName = this.props.index === 0 ? ' start' : '';
+    const modeClassName = 'call';
+
+    return (
+      <div><div key={this.props.index}>
```
Comment (hannesj) [inline app/component/CallAgencyLeg.js]: The two outermost divs are not needed?

**Pull request merged**